packages/lesswrong/components/ea-forum/digest/EditDigestPublishBtn.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { registerComponent } from '../../../lib/vulcan-lib';
+import { useUpdate } from '../../../lib/crud/withUpdate';
+import Button from '@material-ui/core/Button';
+import { useDialog } from '../../common/withDialog';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  publishBtn: {
+    fontSize: 14,
+    textTransform: 'none',
+    boxShadow: 'none'
+  },
+})
+
+
+const EditDigestPublishBtn = ({digest, classes} : {
+  digest: DigestsMinimumInfo,
+  classes: ClassesType
+}) => {
+  const { openDialog } = useDialog()
+  const isPublished = !!digest.publishedDate
+  
+  const { mutate: updateDigest } = useUpdate({
+    collectionName: 'Digests',
+    fragmentName: 'DigestsMinimumInfo',
+  })
+  
+  /**
+   * If the digest has been published before, set or unset the publishedDate.
+   * Otherwise, open the publish confirmation dialog.
+   */
+  const handleBtnClick = () => {
+    // if the digest has an endDate set, then we know it's already been published
+    if (digest.endDate) {
+      void updateDigest({
+        selector: {_id: digest._id},
+        data: {
+          publishedDate: isPublished ? null : new Date()
+        }
+      })
+    } else {
+      openDialog({
+        componentName: 'ConfirmPublishDialog',
+        componentProps: {digest}
+      })
+    }
+  }
+
+  return (
+    <Button
```

Comment:
It's not something to do in this PR but, note to self, we really need a `SubmitButton` component with proper styles because we've got several all over the place that try to look the same but have subtle differences/bugs. (This one has no hover styles in dark mode for instance, which doesn't really matter on an admin page, but it'd be really nice if we just didn't need to think about it).

Reply:
Yeah I think there are a bunch of common components like this that would be nice to have... Maybe next time I want a quick win I'll make one